### PR TITLE
Add Jingle SOCKS5Bytestream transport

### PR DESCRIPTION
**Pull request by vanitasvitae**
This PR adds elements and provider for Jingle Socks5BytestreamTransport (xep-0260)

### CODE REVIEW

File: smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/elements/JingleS5BTransport.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        public Builder setCandidateError() {
+            return addTransportInfo(JingleS5BTransportInfo.CandidateError());
+        }
+
+        public Builder setProxyError() {
+            return addTransportInfo(JingleS5BTransportInfo.ProxyError());
+        }
+
+        public JingleS5BTransport build() {
+            StringUtils.requireNotNullOrEmpty(streamId, "sid MUST be neither null, nor empty.");
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/elements/JingleS5BTransport.java]: verification is usually performed as late as possible, which means in the constructor. not in the build() method

File: smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/elements/JingleS5BTransportCandidate.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            return this;
+        }
+
+        public Builder setType(Type type) {
+            this.type = type;
+            return this;
+        }
+
+        public JingleS5BTransportCandidate build() {
+            Objects.requireNonNull(cid);
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/elements/JingleS5BTransportCandidate.java]: See comment above about verification

File: smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/provider/JingleS5BTransportProvider.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        String mode = parser.getAttributeValue(null, JingleS5BTransport.ATTR_MODE);
+        if (mode != null) {
+            builder.setMode(mode.equals(udp.toString()) ? udp : tcp);
+        }
+
+        JingleS5BTransportCandidate.Builder cb;
+        while (true) {
+            int tag = parser.nextTag();
+            String name = parser.getName();
+            if (tag == START_TAG) {
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/provider/JingleS5BTransportProvider.java]: Always use switch/case here too

File: smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/provider/JingleS5BTransportProvider.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                switch (name) {
+
+                    case JingleS5BTransportCandidate.ELEMENT:
+                    cb = JingleS5BTransportCandidate.getBuilder();
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/transports/jingle_s5b/provider/JingleS5BTransportProvider.java]: Indentation appears to be off for this case.

**Comment by Flowdalic** [comment]
Looks good, squash please.

**Comment by vanitasvitae** [comment]
Squashed

**Pull request merged**